src/endpoints/auth.ts

```diff
@@ -0,0 +1,64 @@
+import { add, getUnixTime } from 'date-fns';
+
+export interface AuthTokenResponse {
+    accessToken: string;
+    credential: Credential;
+    expires: number;
+    IDToken: string;
+    role: string;
+    sub: string;
+}
+
+export interface Credential {
+    iss: string;
+    sub: string;
+    exp: number;
+    ext: Ext;
+}
+
+export interface Ext {
+    avatarURL: string;
+    displayName: string;
+    email: string;
+    firstName: string;
+    lastName: string;
+    locale: string;
+    orgs: string[];
+}
+
+export interface AuthTokenResponseReduced
+    extends Pick<AuthTokenResponse, 'accessToken' | 'expires' | 'IDToken'> {
+    ext: Ext;
+}
```

Comment:
Not sure what we want to call this and I do not care too much. Basically it is for objects that are returned from the server with more stuff than we need... so I called it `Reduced`